### PR TITLE
[Tooling] Add Release Publishing automation

### PR DESCRIPTION
**Pull request by iangmaia**
This PR automates the release publishing, which means to:

- Publish the existing draft release on GitHub (which will also have GitHub create the associated git tag, pointing to the tip of the branch)
- If the release branch for the next version already exists, backmerge the current release into it
- Delete the current release branch

This PR will be followed by updates to the Releases V2 tool.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -537,6 +537,45 @@ platform :ios do
...
   end
 
+  # This lane publishes a release on GitHub, tagging it on git, and backmerges the current release branch into the next release/ branch
+  #
+  # @param [Boolean] skip_confirm (default: false) If set, will skip the confirmation prompt before running the rest of the lane
+  #
+  # @example Running the lane
+  #          bundle exec fastlane publish_release_and_cleanup skip_confirm:true
+  #
+  lane :publish_release_and_cleanup do |skip_confirm: false|
```
Comment (AliSoftware) [inline fastlane/Fastfile]: Same 💭  as https://github.com/woocommerce/woocommerce-android/pull/12450#discussion_r1739151450

Comment (iangmaia) [reply fastlane/Fastfile]: I've updated the PR after the release of [release-toolkit 12.0.0](https://rubygems.org/gems/fastlane-plugin-wpmreleasetoolkit/versions/12.0.0).

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/222">20.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by AliSoftware — APPROVED** [review]

**Pull request merged**